driftbase/parties.py

```diff
@@ -0,0 +1,236 @@
+            pipe.delete(scoped_player_party_key)
+            pipe.delete(sending_player_invites_key)
+            result = pipe.execute()
+            return result
+        except WatchError:
+            abort(http_client.CONFLICT)
+
+
+def disband_party(party_id):
+    scoped_party_players_key = make_party_players_key(party_id)
+
+    with g.redis.conn.pipeline() as pipe:
+        try:
+            pipe.watch(scoped_party_players_key)
+            players = pipe.smembers(scoped_party_players_key)
+            pipe.multi()
+            for player in players:
+                pipe.delete(make_player_party_key(int(player)))
+            pipe.delete(scoped_party_players_key)
+            result = pipe.execute()
+            return result
+        except WatchError:
+            abort(http_client.CONFLICT)
+
+
+def create_party_invite(party_id, sending_player_id, invited_player_id):
+    inviting_player_party_key = make_player_party_key(sending_player_id)
+    invited_player_party_key = make_player_party_key(invited_player_id)
+    scoped_invite_id_key = g.redis.make_key("party_invite:id:")
+    sending_player_invites_key = make_player_invites_key(sending_player_id)
+
+    with g.redis.conn.pipeline() as pipe:
+        try:
+            pipe.watch(invited_player_party_key)
+
+            inviting_player_party_id = pipe.get(inviting_player_party_key)
+            if inviting_player_party_id:
+                party_players_key = make_party_players_key(int(inviting_player_party_id))
+                pipe.multi()
+                pipe.sismember(party_players_key, invited_player_id)
+                pipe.get(invited_player_party_key)
+                is_already_in_team, invited_player_party_id = pipe.execute()
+                pipe.watch(invited_player_party_key)
+                if is_already_in_team and invited_player_party_id == inviting_player_party_id:
+                    log.debug("Player {} is already a member of party {}".format(invited_player_id, party_id))
+                    return None
+
+            invite_id = pipe.incr(scoped_invite_id_key)
+            scoped_invite_key = make_party_invite_key(invite_id)
+            pipe.multi()
+            pipe.hset(scoped_invite_key, mapping={b'from': sending_player_id, b'to': invited_player_id})
+            pipe.zadd(sending_player_invites_key, mapping={invite_id: invited_player_id})
+            pipe.execute()
+            return invite_id
+        except WatchError:
+            abort(http_client.CONFLICT)
+
+
+def get_party_invite(invite_id):
+    scoped_party_invite_key = make_party_invite_key(invite_id)
+    return g.redis.conn.hgetall(scoped_party_invite_key)
+
+
+def decline_party_invite(invite_id, declining_player_id):
+    scoped_player_party_key = make_player_party_key(declining_player_id)
+    scoped_party_invite_key = make_party_invite_key(invite_id)
+
+    with g.redis.conn.pipeline() as pipe:
+        try:
+            pipe.watch(scoped_player_party_key)
```

Comment:
Not sure I'm following here; why does this key need to be watched?  Are you guarding against the case where the player joins the party while this method is running?

Reply:
This is probably wrong. I'm also considering just replacing most of these watches with locks

Reply:
I'll remove that watch, can't think of anything that really need to be watched for this operation